sir_cake/store/views.py

```diff
@@ -247,6 +249,7 @@ def shipping(request):
         valid_form_name = _save_name_data(request, name_form)
         if valid_form_address and valid_form_name:
             return redirect('payment')
+    context['expire_date'] = _get_basket_expire_date(context['preorder'])
```

Comment:
I don't see where you insert a `preorder` here in this view? Does this work/break?